File: backend/Margorak.Api/Dto/SaveCharacterDto.cs
namespace Margorak.Api.Dto
{
    public class SaveCharacterDto
    {
        public LocationDto Location { get; set; } = null!;
        public EquippedItemDto[] EquippedItems { get; set; } = [];
        public InventoryItemDto[] InventoryItems { get; set; } = [];
    }
}

File: frontend/src/app/features/character/dto/save-character.dto.ts
import { InventoryItemDto } from '../../../shared/dto/inventory-item.dto';
import { CharacterStatsDto } from './character.dto';
import { EquippedItemDto } from './equipped-item.dto';
import { LocationDto } from './location.dto';

export interface SaveCharacterDto {
  location: LocationDto;
  equippedItems: EquippedItemDto[];
  inventoryItems: InventoryItemDto[];
  characterStats: CharacterStatsDto;
}
